react/components/FoldableContainer.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FunctionComponent } from 'react'
+import LazyRender from './LazyRender'
+
+interface Props {
+  foldIndex?: number
+}
+const FoldableContainer: FunctionComponent<Props> = ({
+  children,
+  foldIndex,
+}) => {
+  if (foldIndex == null || foldIndex === -1) {
+    return <>{children}</>
+  }
+  const childrenArray = React.Children.toArray(children)
+  const aboveTheFold = childrenArray.slice(0, foldIndex)
```

Comment:
Suggesting all of this because if index is zero the above the fold would be an empty array.

Reply:
not sure if I get it--i mean, if you put the fold at the very top of the page, where its index would be `0`, thats the expected behaviour I think: the above-the-fold array would be empty, and all the elements would be contained in the below-the-fold array.

right? not sure if i'm missing something

Reply:
Hmm, ok, I didn't think about that use case (it's kind of weird though)

Reply:
it is probably not a very good use case indeed haha but it would be the expected behaviour of doing so